scrutiny/cli/commands/elf2varmap.py

```diff
@@ -46,20 +47,24 @@ def run(self) -> Optional[int]:
                                          ignore_cu_patterns=args.cu_ignore_patterns,
                                          path_ignore_patterns=args.path_ignore_patterns,
                                          )
-        varmap = extractor.get_varmap()
+        try:
+            varmap = extractor.get_varmap()
 
-        if args.output is None:
-            print(varmap.get_json())
-        else:
-            if os.path.isdir(args.output):
-                output_file = os.path.join(args.output, 'varmap.json')
+            if args.output is None:
+                print(varmap.get_json())
             else:
-                output_file = args.output
+                if os.path.isdir(args.output):
+                    output_file = os.path.join(args.output, 'varmap.json')
+                else:
+                    output_file = args.output
 
-            if os.path.isfile(output_file):
-                logging.warning('File %s already exist. Overwritting' % output_file)
+                if os.path.isfile(output_file):
+                    logging.warning('File %s already exist. Overwritting' % output_file)
 
-            varmap.write(output_file, indent=args.indent)
-            self.getLogger().info(f"Varmap file {output_file} written")
+                varmap.write(output_file, indent=args.indent)
+                self.getLogger().info(f"Varmap file {output_file} written")
+        except Exception as e:
```

Comment:
Probably want to catch OSError exception here, not all Exception. this hides bugs you want to see happen because they should get fixed. ie: errors like empty array handling errors  and divide by zero error.

Reply:
Error are logged in the exception handler, they are not masked. 
If not catched here, they go to the upper level (CLI) where again, it can be logged or reraised to the OS.

Problem is: CMake doesn't print much details in case of failure. You just get a generic failure message. With this, you get more details on what failed.